core/listener_test.go

```diff
@@ -90,11 +90,11 @@ func TestListenerWithWrongChainRPC(t *testing.T) {
 
 	// create Listener and start listening
 	cl := createListener(ctx, t, fetcher, ps0, eds, store, "wrong-chain-rpc")
-	sub, err := cl.fetcher.SubscribeNewBlockEvent(ctx)
+	require.NoError(t, cl.fetcher.Start(ctx))
+	sub, err := cl.fetcher.runSubscriber()
 	require.NoError(t, err)
 
-	err = cl.listen(ctx, sub)
-	assert.ErrorIs(t, err, errInvalidSubscription)
+	assert.Panics(t, func() { cl.listen(ctx, sub) })
```

Comment:
It might be annoying to test the log.Fatal bc it's an os exit status 1 

I guess maybe we can just keep it as a panic for now.